src/multisig/multisigTransactionsFactory.ts

```diff
@@ -0,0 +1,509 @@
+
+    /**
+     * Proposes adding a new proposer
+     */
+    createTransactionForProposeAddProposer(sender: Address, options: resources.ProposeAddProposerInput): Transaction {
+        const dataParts = [
+            "proposeAddProposer",
+            this.argSerializer.valuesToStrings([new AddressValue(options.proposerAddress)])[0],
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: sender,
+            receiver: options.multisigContract,
+            dataParts: dataParts,
+            gasLimit: options.gasLimit,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    /**
+     * Proposes removing a user (board member or proposer)
+     */
+    createTransactionForProposeRemoveUser(sender: Address, options: resources.ProposeRemoveUserInput): Transaction {
+        const dataParts = [
+            "proposeRemoveUser",
+            this.argSerializer.valuesToStrings([new AddressValue(options.userAddress)])[0],
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: sender,
+            receiver: options.multisigContract,
+            dataParts: dataParts,
+            gasLimit: options.gasLimit,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    /**
+     * Proposes changing the quorum (minimum signatures required)
+     */
+    createTransactionForProposeChangeQuorum(sender: Address, options: resources.ProposeChangeQuorumInput): Transaction {
+        const dataParts = [
+            "proposeChangeQuorum",
+            this.argSerializer.valuesToStrings([new U32Value(options.newQuorum)])[0],
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: sender,
+            receiver: options.multisigContract,
+            dataParts: dataParts,
+            gasLimit: options.gasLimit,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    /**
+     * Proposes a transaction that will transfer EGLD and/or execute a function
+     */
+    createTransactionForProposeTransferExecute(
+        sender: Address,
+        options: resources.ProposeTransferExecuteInput,
+    ): Transaction {
+        const gasOption = new U64Value(options.gasLimit);
+        const input = resources.ProposeTransferExecutInput.newFromTransferExecuteInput({
+            multisig: options.multisigContract,
+            to: options.to,
+            tokenTransfers: [],
+            functionName: options.functionName,
+            arguments: options.functionArguments,
+            abi: options.abi,
+        });
+        const dataParts = [
+            "proposeTransferExecute",
+            this.argSerializer.valuesToStrings([new AddressValue(options.to)])[0],
+            this.argSerializer.valuesToStrings([new BigUIntValue(options.egldAmount)])[0],
+            this.argSerializer.valuesToStrings([new OptionValue(new OptionType(new U64Type()), gasOption)])[0],
```

Comment:
can you perhaps call `valuesToStrings()` only once and pass all the arguments at once?